plugins/kubernetes/app/assets/javascripts/kubernetes/factories/build_factory.js

```diff
@@ -0,0 +1,16 @@
+samson.factory('buildFactory', function() {
```

Comment:
I know it seems quite a java-like pattern, but people usually say that a lot about AngularJS in general. :)

I've researched quite a bit around how to build proper model objects in Javascript with AngularJS, and all the references I saw mentioned using a Factory to build the object. That's why I went that way.

The advantages I see of using this approach instead of pure JSON objects is that you can leverage the model to support read-only objects (Object.freeze()) and rich client-side validations. I'm still not doing it, though, but I'll probably start doing it soon.

Just a few references:
https://medium.com/opinionated-angularjs/angular-model-objects-with-javascript-classes-2e6a067c73bc
http://www.webdeveasy.com/angularjs-data-model/
https://www.mutuallyhuman.com/blog/2014/05/08/angularjs-services-and-factories-done-right/